gallery/gallery/lib/demos/cupertino/cupertino_tab_bar_demo.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+// BEGIN cupertinoNavigationDemo
+
+class _TabInfo {
+  const _TabInfo(this.title, this.icon);
+
+  final String title;
+  final IconData icon;
+}
+
+class CupertinoTabBarDemo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final _tabInfo = <_TabInfo>[
+      _TabInfo(
+        GalleryLocalizations.of(context).cupertinoTabBarHomeTab,
+        CupertinoIcons.home,
+      ),
+      _TabInfo(
+        GalleryLocalizations.of(context).cupertinoTabBarChatTab,
+        CupertinoIcons.conversation_bubble,
+      ),
+      _TabInfo(
+        GalleryLocalizations.of(context).cupertinoTabBarProfileTab,
+        CupertinoIcons.profile_circled,
+      ),
+    ];
+
+    return DefaultTextStyle(
+      style: CupertinoTheme.of(context).textTheme.textStyle,
+      child: CupertinoTabScaffold(
+        tabBar: CupertinoTabBar(
+          items: [
+            for (final tabInfo in _tabInfo)
+              BottomNavigationBarItem(
+                title: Text(tabInfo.title),
+                icon: Icon(tabInfo.icon),
+              ),
+          ],
+        ),
+        tabBuilder: (context, index) {
+          switch (index) {
+            case 0:
```

Comment:
Looks like you might be able to simplify this switch statement with:

```dart
tabBuilder: (context, index) {
  return CupertinoTabView(
    builder: (context) => _CupertinoDemoTab(
      title: _tabInfo[index].title,
      icon: _tabInfo[index].icon,
    ),
    defaultTitle: _tabInfo[index].title,
  );
}
```

Reply:
Done